library/blivet.py

```diff
@@ -1815,9 +1902,104 @@ def _look_up_device(self):
             self._thinpools = [d for d in self._device.children if d.type == 'lvmthinpool']
 
 
+class BlivetStratisPool(BlivetPool):
+    blivet_device_class = devices.StratisPoolDevice
+
+    def _type_check(self):
+        return self._device.type == "stratis pool"
+
+    def _update_from_device(self, param_name):
+        """ Return True if param_name's value was retrieved from a looked-up device. """
+        log.debug("BlivetStratisPool._update_from_device: %s", self._device)
+
+        if param_name == 'disks':
+            self._pool['disks'] = [d.name for d in self._device.disks]
+        elif param_name == 'encryption':
+            self._pool['encryption'] = self._device.encrypted
+        elif param_name == 'encryption_clevis_pin':
+            self._pool['encryption_clevis_pin'] = self._device._clevis.pin
```

Comment:
ok - I see - the name "pin" was chosen because `_device._clevis.pin` is the field name.  Still, I think the storage role should use a more descriptive parameter name (unless this really is a PIN and not the type of clevis to use).